CMakeLists.txt

```diff
@@ -951,59 +951,59 @@ if(ENABLE_ZERO_LENGTH_COORD_BOUND)
 endif()
 
 # Enable Parallel Tests.
-option(ENABLE_PARALLEL_TESTS "Enable Parallel IO Tests. Requires HDF5/NetCDF4 with parallel I/O Support." "${HDF5_PARALLEL}")
-if(ENABLE_PARALLEL_TESTS AND USE_PARALLEL)
+option(NETCDF_ENABLE_PARALLEL_TESTS "Enable Parallel IO Tests. Requires HDF5/NetCDF4 with parallel I/O Support." "${HDF5_PARALLEL}")
+if(NETCDF_ENABLE_PARALLEL_TESTS AND USE_PARALLEL)
   set(TEST_PARALLEL ON CACHE BOOL "")
   if(USE_NETCDF4)
     set(TEST_PARALLEL4 ON CACHE BOOL "")
   endif()
 endif()
 
-IF (ENABLE_PARALLEL_TESTS AND NOT USE_PARALLEL)
+IF (NETCDF_ENABLE_PARALLEL_TESTS AND NOT USE_PARALLEL)
   message(FATAL_ERROR "Parallel tests requested, but no parallel HDF5 installation detected.")
 endif()
 
 # Enable special filter test; experimental when using cmake.
-option(ENABLE_FILTER_TESTING "Enable filter testing. Ignored if shared libraries or netCDF4 are not enabled" ${ENABLE_PLUGINS})
+option(NETCDF_ENABLE_FILTER_TESTING "Enable filter testing. Ignored if shared libraries or netCDF4 are not enabled" ${NETCDF_ENABLE_PLUGINS})
 
-if(ENABLE_FILTER_TESTING)
-  if(NOT ENABLE_HDF5 AND NOT ENABLE_NCZARR)
-    message(WARNING "ENABLE_FILTER_TESTING requires HDF5 and/or NCZarr. Disabling.")
-    set(ENABLE_FILTER_TESTING OFF CACHE BOOL "Enable Filter Testing" FORCE)
+if(NETCDF_ENABLE_FILTER_TESTING)
+  if(NOT NETCDF_ENABLE_HDF5 AND NOT NETCDF_ENABLE_NCZARR)
+    message(WARNING "NETCDF_ENABLE_FILTER_TESTING requires HDF5 and/or NCZarr. Disabling.")
+    set(NETCDF_ENABLE_FILTER_TESTING OFF CACHE BOOL "Enable Filter Testing" FORCE)
   endif()
 endif()
 
 if(NOT BUILD_SHARED_LIBS)
```

Comment:
Can add `NETCDF_SHARED_LIBS` option and demote `BUILD_SHARED_LIBS` to a local variable? This would make it easier to control in fetch-content environment.